File: server/models/category.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime

class Category(BaseModel):
    id: Optional[str]  # MongoDB ObjectId
    name: str
    slug: str
    description: Optional[str]
    parent: Optional[str]  # Reference to parent Category ID
    image: Optional[str]
    color: Optional[str]
    meta_title: str = ""
    meta_description: str = ""
    meta_keywords: str = ""
    subcategories: List[str] = []  # References to subcategory IDs
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

File: server/models/Category.js
const mongoose = require('mongoose');

const categorySchema = new mongoose.Schema({
  name: { type: String, required: true, unique: true },
  slug: { type: String, required: true, unique: true },
  description: { type: String },
  parent: { type: mongoose.Schema.Types.ObjectId, ref: 'Category', default: null },
  image: { type: String },
  color: { type: String },
  // SEO Metadata
  metaTitle: { type: String, default: '' },
  metaDescription: { type: String, default: '' },
  metaKeywords: { type: String, default: '' },
}, { timestamps: true, toJSON: { virtuals: true, versionKey: false }, toObject: { virtuals: true } });

// Virtual for subcategories
categorySchema.virtual('subcategories', {
  ref: 'Category',
  localField: '_id',
  foreignField: 'parent',
  justOne: false,
});

const Category = mongoose.model('Category', categorySchema);
module.exports = Category;

